src/uartdef.h

```diff
@@ -0,0 +1,69 @@
+#include <stdio.h>
+#include <sys/file.h>
+#include <termios.h>
+
+#pragma once
+
+#define DEVICE_PATH_SIZE 128
+
+#define APBUART_NUM 6
+#define APBUART_BUFFER_SIZE 1024
+#define APB_START 0x80000000
+#define APBUART_ADDR_MASK 0xFFFF
+#define APBUART_REGISTER_TYPE_MASK 0xFF
+
+#define APBUART0_START_ADDRESS 0x80000100
+#define APBUART1_START_ADDRESS 0x80100100
+#define APBUART2_START_ADDRESS 0x80100200
+#define APBUART3_START_ADDRESS 0x80100300
+#define APBUART4_START_ADDRESS 0x80100400
+#define APBUART5_START_ADDRESS 0x80100500
+
+#define APBUART0_IRQ 2
+#define APBUART1_IRQ 17
+#define APBUART2_IRQ 18
+#define APBUART3_IRQ 19
+#define APBUART4_IRQ 20
+#define APBUART5_IRQ 21
+
+typedef struct
+{
+    FILE *file;
+    int descriptor;
+    char buffer[APBUART_BUFFER_SIZE];
+    int buffer_size_cnt;
```

Comment:
Why cnt? Isn't this just buffer_size? Same for index... isn't it just buffer_index?